Content.Client/Doors/DoorSystem.Airlock.cs

```diff
@@ -0,0 +1,152 @@
+using Content.Client.Wires.Visualizers;
+using Content.Shared.Doors.Components;
+using Content.Shared.Doors.Systems;
+using Robust.Client.Animations;
+using Robust.Client.GameObjects;
+
+namespace Content.Client.Doors;
+
+public sealed partial class DoorSystem
+{
+    [Dependency] private readonly AppearanceSystem _appearanceSystem = default!;
+
+    private void InitializeClientAirlock()
+    {
+        SubscribeLocalEvent<AirlockComponent, ComponentStartup>(OnComponentStartup);
+        SubscribeLocalEvent<AirlockComponent, AppearanceChangeEvent>(OnAppearanceChange);
+    }
+
+    private void OnComponentStartup(Entity<AirlockComponent> airlock, ref ComponentStartup args)
+    {
+        // Has to be on component startup because we don't know what order components initialize in and running this before DoorComponent inits _will_ crash.
+        if (!TryComp<DoorComponent>(airlock, out var door))
+            return;
+
+        if (airlock.Comp
+            .OpenUnlitVisible) // Otherwise there are flashes of the fallback sprite between clicking on the door and the door closing animation starting.
+        {
```

Comment:
```suggestion
        if (airlock.Comp.OpenUnlitVisible)
        {
            // Otherwise there are flashes of the fallback sprite between clicking on the door and the door closing animation starting.
```